File: C++/Euler (C++)/src/base/problem.cpp
// abstract class for problems
#ifndef __PROBLEM_H__
#include "problem.h"
#endif

#include <iostream>   // cout
#include "timer.h"
#include "config.h"

void Problem::run(int variation)
{
   Timer* t = new Timer();
   t->start();

   std::string answer = execute(variation);

   t->stop();
   executionTimeNs = t->getDurationNs();

   if (answer != Config::Answer(problem_number))
   {
      std::cout << "---Problem: " << problem_number;
      if (variation >= 0)
         std::cout << "_" << variation << " ";
      std::cout << " ::: Expected (" << Config::Answer(problem_number) << ") but got (" << answer << ")" << std::endl;
   }
}

Problem::~Problem()
{
}

File: Java/Euler (Java)/src/base/Problem.java
package base;

import util.Stopwatch;


public abstract class Problem {
	
	protected abstract String execute(int variation);
	protected int problemNumber = Integer.valueOf(getClass().getSimpleName().substring(1));
	protected int variations;
	protected long executeTimeNs;
	
	public void run(int variation)
	{
		Stopwatch s = new Stopwatch();
		s.start();
		
		String answer = execute(variation);
		
		executeTimeNs = s.stop();
		if (!answer.equals(Config.INSTANCE.answer(problemNumber)))
		{
			System.out.print("---Problem: " + problemNumber);
			if (variation >= 0)
				System.out.print("_" + variation);
			System.out.println(" Expected (" + Config.INSTANCE.answer(problemNumber) + ") but got (" + answer + ")");
		} 
	}
	
	public int getProblemNumber()
	{
		return problemNumber;
	}
	
	public int getVariations()
	{
		return variations;
	}
	
	public long getExecuteTimeNs()
	{
		return executeTimeNs;
	}

}
